experimental/PartitionedPermutations/src/PartitionedPermutation.jl

```diff
@@ -0,0 +1,107 @@
+"""
+    PartitionedPermutation
+
+The type of partitioned permutations. Fieldnames are
+- p::Perm{Int} - a permutation
+- V::SetPartition - a partition
+If the permutation has length `n`, then the partition must have `n` upper points and 0 lower points. 
+Further, if `W` is the partition given by the cycles of `p`, then `W` must be dominated by `V` in the 
+sense that every block of `W` is contained in one block of `V`. There is one inner constructer of PartitionedPermutation:
+- PartitionedPermutation(p::Perm{Int}, V::Vector{Int}) constructs the partitioned permutation where the partition is given by the vector V.
+If the optional flag `check` is set to `false`, then the constructor skips the validation of the requirements mentioned above.
+
+# Examples
+```jldoctest
+julia> PartitionedPermutation(Perm([2, 1, 3]), [1, 1, 2])
+PartitionedPermutation((1,2), SetPartition([1, 1, 2], Int64[]))
+```
+"""
+struct PartitionedPermutation
+    p::Perm{Int}
+    V::SetPartition
+
+    function PartitionedPermutation(p::Perm{Int}, V::Vector{Int}; check::Bool=true)
+        _V = set_partition(V, Int[])
+        if check
+            @req parent(p).n == length(V) "permutation and partition must have the same length"
+            @req is_dominated_by(cycle_partition(p), _V) "permutation must be dominated by partition"
+        end
+        new(p, _V)
+    end
+end
+
+
+"""
+    partitioned_permutation(p::Perm{Int}, V::Vector{Int}, check::Bool=true)
+
+Construct and output a `PartitionedPermutation`.
+
+# Examples
+```jldoctest
+julia> partitioned_permutation(Perm([2, 1, 3]), [1, 1, 2])
+PartitionedPermutation((1,2), SetPartition([1, 1, 2], Int64[]))
+```
+"""
+function partitioned_permutation(p::Perm{Int}, V::Vector{Int}; check::Bool=true)
+    return PartitionedPermutation(p, V; check)
+end
+
+function ==(pp_1::PartitionedPermutation, pp_2::PartitionedPermutation)
+    return pp_1.p == pp_2.p && pp_1.V == pp_2.V
+end
+
+function hash(pp::PartitionedPermutation, h::UInt)
+    return hash(pp.p, hash(pp.V, h))
+end
+
+function deepcopy_internal(pp::PartitionedPermutation, stackdict::IdDict)
+    if haskey(stackdict, pp)
+        return stackdict[pp]
+    end
+    q = PartitionedPermutation(deepcopy_internal(pp.p, stackdict), 
+                     deepcopy_internal(pp.V, stackdict))
+    stackdict[pp] = q
+    return q
+end
+
+function get_partition(pp::PartitionedPermutation)
```

Comment:
This name is rather unfortunate given that `set_partition` is not the corresponding setter but does something completely different.
Furthermore, in the usual julia style, one would name this function `partition(::PartitionedPermutation)`